pkg/activator/net/revision_backends.go

```diff
@@ -158,6 +158,11 @@ func (rw *revisionWatcher) probeClusterIP(dest string) (bool, error) {
 // probePodIPs will probe the given target Pod IPs and will return
 // the ones that are successfully probed, or an error.
 func (rw *revisionWatcher) probePodIPs(dests []string) (sets.String, error) {
+	// Short circuit case where dests == healthyPods
+	if rw.allDestsHealthy(dests) {
+		return rw.healthyPods, nil
```

Comment:
Could this be `sets.NewString(dests)` and change 268 to `len(dests) <= len(rw.healthyPods)`?

Reply:
That defeats the purpose of this PR not to allocate more memory :-)